packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -150,7 +190,26 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   /// See also:
   ///
   ///  * [AnnotatedRegionLayer], for placing values in the layer tree.
-  Iterable<S> findAll<S>(Offset regionOffset);
+  Iterable<S> findAll<S>(Offset regionOffset) {
```

Comment:
seems weird that findAll and find are not diverged in implementation so much. could one be implemented in terms of the other?

Reply:
Originally find was re-implemented in terms of findAll in @gspencergoog 's original PR, but that end up regressing several benchmarks. The performance might have been fixed since then

Reply:
i was thinking more the other way around (findAll is by design more expensive than find)

Reply:
I thought for a very long time and see if there was a way that could make `find` and `findAll` share code with methods such as `visitChildren` and `visitSelf`. I could not come up with a way without introducing several ridiculously complex callbacks. I've decided to add another parameter `bool onlyFirst`.